src/python/WMCore/Services/LogDB/LogDB.py

```diff
@@ -126,5 +127,55 @@ def cleanup(self, thr, backend='local'):
         try:
             self.backend.cleanup(thr)
         except Exception as exc:
-            self.logger.error('LogDBBackend cleanup API failed, backend=%s, error=%s' \
-                    % (backend, str(exc)))
+            self.logger.error('LogDBBackend cleanup API failed, backend=%s, error=%s', backend, str(exc))
+    
+    def heartbeat_report(self):
+        report = defaultdict(dict)
+        if self.user_pat.match(self.identifier):
+            self.logger.error("User %s: doesn't allow this function", self.identifier) 
+            return report
+        
+        for row in self.backend.get(self.default_user, None, agent=True).get('rows', []):
+            identifier = row['doc']['identifier']
+            # wmstats thread can run in multiple boxed. 
+            if self.identifier == identifier or identifier.startswith(self.identifier):
+                #this will handle wmstats DataCacheUpdate thread with multiple machine
+                postfix = identifier.replace(self.identifier, "")
+                thr = "%s%s" % (row['doc']['thr'], postfix)
+                mtype = row['doc']['type']
+                ts = row['doc']['messages'][-1]['ts']
+                msg = row['doc']['messages'][-1]['msg']
+                if (thr in report) and ('ts' in report[thr]) and ts <= report[thr]['ts']:
+                    continue
+                else:
+                    report[thr]['type'] = mtype
+                    report[thr]['msg'] = msg
+                    report[thr]['ts'] = ts
+        return report
+    
+    def _append_down_component_detail(self, report, thr, msg, ts=0, state="error"):
```

Comment:
If thread crashes we always have the record. if thread never started we won't know. I don't think it makes sense to add time (current time) if the thread never started.